var/spack/repos/builtin/packages/py-netcdf4/package.py

```diff
@@ -36,29 +37,30 @@ class PyNetcdf4(PythonPackage):
     depends_on("py-setuptools-scm@3.4:+toml", when="@1.7:", type="build")
     depends_on("py-cftime", type=("build", "run"))
     depends_on("py-certifi", when="@1.6.5:", type=("build", "run"))
-    depends_on("py-numpy", when="@1.6.5:", type=("build", "link", "run"))
+    depends_on("py-numpy", type=("build", "link", "run"))
+    depends_on("py-numpy@2.0:", when="@1.7.1:", type=("build", "link", "run"))
     depends_on("py-numpy@1.9:", when="@1.5.4:1.6.2", type=("build", "link", "run"))
-    depends_on("py-numpy@1.7:", type=("build", "link", "run"))
     # https://github.com/Unidata/netcdf4-python/pull/1317
     depends_on("py-numpy@:1", when="@:1.6", type=("build", "link", "run"))
     depends_on("py-mpi4py", when="+mpi", type=("build", "run"))
-    depends_on("netcdf-c", when="-mpi")
+    depends_on("netcdf-c~mpi", when="-mpi")  # netcdf-c has +mpi by default
     depends_on("netcdf-c+mpi", when="+mpi")
-    depends_on("hdf5@1.8.0:+hl", when="-mpi")
+    depends_on("hdf5@1.8.0:+hl~mpi", when="-mpi")  # hdf5 defaults to +mpi by default
     depends_on("hdf5@1.8.0:+hl+mpi", when="+mpi")
 
     # The installation script tries to find hdf5 using pkg-config. However, the
     # version of hdf5 installed with Spack does not have pkg-config files.
     # Therefore, if pkg-config finds hdf5.pc at all (e.g. provided by
     # Ubuntu/Debian package manager), it is definitely not what we need. The
     # following patch disables the usage of pkg-config at all.
-    patch("disable_pkgconf.patch")
+    patch("disable_pkgconf.patch", when="@:1.6.5")
+    patch("disable_pkgconf-1.7.patch", when="@1.7.1.post2:")
```

Comment:
I have just built version `1.7.1.post2` with the existing patch. It also seems to work for the current [`master`](https://github.com/Unidata/netcdf4-python/tree/9426a684d3bd7b6c0c9b4d3c3f8531f30583d1e2) of the package.

Reply:
Interesting. On both a mac and linux machine I was getting errors about patches being already applied.

Reply:
I can reproduce your finding, and I have no explanation. I've removed the version specific patch. If it shows up as a problem we can add it back later.